pageserver/src/tenant/vectored_blob_io.rs

```diff
@@ -76,14 +76,115 @@ pub(crate) enum VectoredReadExtended {
     No,
 }
 
-pub(crate) struct VectoredReadBuilder {
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+pub enum VectoredReadCoalesceMode {
+    /// Only coalesce exactly adjacent reads.
+    AdjacentOnly,
+    /// In addition to adjacent reads, also consider reads whose corresponding
+    /// `end` and `start` offsets reside at the same chunk.
+    Chunked(usize),
+}
+
+impl VectoredReadCoalesceMode {
+    pub(crate) fn get() -> Self {
+        let align = virtual_file::get_io_buffer_alignment();
+        if align == 1 {
+            VectoredReadCoalesceMode::AdjacentOnly
+        } else {
+            VectoredReadCoalesceMode::Chunked(align)
+        }
+    }
+}
+
+pub(crate) enum VectoredReadBuilder {
+    Adajacent(AdjacentVectoredReadBuilderInner),
+    Chunked(ChunkedVectoredReadBuilderInner),
+}
```

Comment:
Why the `Inner` suffix on these types? You already have the `Adjacent|Chunked` prefix.

Also, typo:


```suggestion
pub(crate) enum VectoredReadBuilder {
    Adjacent(AdjacentVectoredReadBuilderInner),
    Chunked(ChunkedVectoredReadBuilderInner),
}
```

Reply:
d0fd9b5d4a496d6c40687bfd6ba114ed5213bf86